asset/asset.go

```diff
@@ -462,6 +464,136 @@ const (
 	ScriptV0 ScriptVersion = 0
 )
 
+// TapscriptTreeNodes represents the two supported ways to define a tapscript
+// tree to be used as a sibling for a Taproot Asset commitment, an asset group
+// key, or an asset script key. This type is used for interfacing with the DB,
+// not for supplying in a proof or key derivation. The inner fields are mutually
+// exclusive.
+type TapscriptTreeNodes struct {
+	// An ordered list of TapLeaf objects, representing a Tapscript tree.
+	leaves *TapLeafNodes
+
+	// The TapHashes of the child nodes of a TapBranch.
+	branch *TapBranchNodes
+}
+
+type TapLeafNodes struct {
+	v []txscript.TapLeaf
```

Comment:
Could this instead be a direct type alias for a slice? E.g. `type TapLeafNodes []txscript.TapLeaf`?

Reply:
I think in that case we wouldn't be able to enforce using the constructors or decoders in `asset` as the only way to create an instance of `TapLeafNodes`.